client/src/partials/primary_cash/income/transfer/transfer.js

```diff
@@ -138,6 +141,23 @@ angular.module('bhima.controllers')
       .finally();
     };
 
+    function haltOnNoExchange () {
+      if (exchange.hasDailyRate()) { return; }
+
+      var instance = $modal.open({
+        templateUrl : 'partials/exchangeRateModal/exchangeRateModal.html',
+        backdrop    : 'static',
+        keyboard    : false,
+        controller  : 'exchangeRateModal'
+      });
+      
+      instance.result.then(function () {
+        $location.path('/exchange_rate');
```

Comment:
Sorry, I have not found a better way to put $location.path('/exchange_rate')
in the controller

2015-09-03 22:09 GMT+01:00 Jonathan Niles <notifications@github.com>:

> In client/src/partials/primary_cash/income/transfer/transfer.js
> <https://github.com/IMA-WorldHealth/bhima/pull/676#discussion_r38697045>:
>
> > @@ -138,6 +141,23 @@ angular.module('bhima.controllers')
> >        .finally();
> >      };
> >
> > +    function haltOnNoExchange () {
> > +      if (exchange.hasDailyRate()) { return; }
> > +
> > +      var instance = $modal.open({
> > +        templateUrl : 'partials/exchangeRateModal/exchangeRateModal.html',
> > +        backdrop    : 'static',
> > +        keyboard    : false,
> > +        controller  : 'exchangeRateModal'
> > +      });
> > +
> > +      instance.result.then(function () {
> > +        $location.path('/exchange_rate');
>
> Since all of the call $location.path('/exchange_rate') on exit, I would
> put that in the exchangeRateModalController.
>
> These changes look great!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/IMA-WorldHealth/bhima/pull/676/files#r38697045>.
>